File: deno/src/wxwork/wxw_usr/wxw_usr.resolver.ts
import {
  useContext,
} from "/lib/context.ts";

import type {
  WxwLoginByCodeInput,
} from "/gen/types.ts";

/**
 * 通过host获取appid, agentid
 * @param host 域名
 */
export async function wxwGetAppid(
  host: string,
) {
  const {
    wxwGetAppid,
  } = await import("./wxw_usr.service.ts");
  const context = useContext();
  
  context.notVerifyToken = true;
  
  const res = await wxwGetAppid(host);
  return res;
}

/**
 * 企业微信单点登录
 */
export async function wxwLoginByCode(
  input: WxwLoginByCodeInput,
) {
  const {
    wxwLoginByCode,
  } = await import("./wxw_usr.service.ts");
  const context = useContext();
  
  context.notVerifyToken = true;
  const res = await wxwLoginByCode(input);
  return res;
}

/**
 * 企业微信同步企微用户
 */
export async function wxwSyncUsr(
  host: string,
) {
  const {
    wxwSyncUsr,
  } = await import("./wxw_usr.service.ts");
  await wxwSyncUsr(host);
  return 0;
}

File: rust/src/src/wxwork/wxw_usr/wxw_usr_resolver.rs
use color_eyre::eyre::Result;

use super::wxw_usr_service;

use super::wxw_usr_model::{
  WxwGetAppid,
  WxwLoginByCodeInput,
  WxwLoginByCode,
};

/// 通过host获取appid, agentid
pub async fn wxw_get_appid(
  host: String,
) -> Result<WxwGetAppid> {
  
  let res = wxw_usr_service::wxw_get_appid(
    host,
  ).await?;
  
  Ok(res)
}

/// 微信企业号登录
pub async fn wxw_login_by_code(
  input: WxwLoginByCodeInput,
) -> Result<WxwLoginByCode> {
  
  let res = wxw_usr_service::wxw_login_by_code(
    input,
  ).await?;
  
  Ok(res)
}

/// 同步企业微信用户
pub async fn wxw_sync_usr(
  host: String,
) -> Result<i32> {
  
  let res = wxw_usr_service::wxw_sync_usr(
    host,
  ).await?;
  
  Ok(res)
}
